doc/build/dts/troubleshooting.rst

```diff
@@ -29,6 +31,179 @@ Many Zephyr header files rely on information from devicetree, so including some
 other API may transitively include :file:`devicetree.h`, but that's not
 guaranteed.
 
+undefined reference to ``__device_dts_ord_<N>``
+***********************************************
+
+This usually happens on a line like this:
+
+.. code-block:: c
+
+   const struct device *dev = DEVICE_DT_GET(NODE_ID);
+
+where ``NODE_ID`` is a valid :ref:`node identifier <dt-node-identifiers>`, but
+no device driver has allocated a ``struct device`` for this devicetree node.
+You thus get a linker error, because you're asking for a pointer to a device
+that isn't defined.
+
+To fix it, you need to make sure that:
+
+1. The node is enabled: the node must have ``status = "okay";``.
+
+   (Recall that a missing ``status`` property means the same thing as ``status
+   = "okay";``; see :ref:`dt-important-props` for more information about
+   ``status``).
+
+2. A device driver responsible for allocating the ``struct device`` is enabled.
+   That is, the Kconfig option which makes the build system compile the driver
+   sources into your application needs to be set to ``y``.
+
+   (See :ref:`setting_configuration_values` for more information on setting
+   Kconfig options.)
+
+**Making sure the node is enabled**:
+
+To find the devicetree node you need to check, use the number ``<N>`` from the
+linker error. Look for this number in the list of nodes at the top of
+:file:`build/zephyr/include/generated/devicetree_generated.h`. For example, if
```

Comment:
build assumes that they named the build directory build but that's not a given, would prefer something like "from the build directory..."